File: app/src/main/java/com/example/android/journalapp/utilities/Journal.java
package com.example.android.journalapp.utilities;

import java.util.Date;

public class Journal {
    private String title;
    private String body;
    private Date date;

    public Journal(){}

    public Journal(String title, String body, Date date){
        this.title = title;
        this.body = body;
        this.date = date;
    }

    public String getTitle(){
        return title;
    }

    public String getBody(){
        return body;
    }

    public Date getDate(){
        return date;
    }
}

File: app/src/main/java/com/journalingapp/model/Journal.kt
package com.journalingapp.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.io.Serializable
import java.util.Date

@Entity(tableName = "journals")
data class Journal(
    @PrimaryKey(autoGenerate = true) var id: Long = 0,
    @ColumnInfo(name = "title") var title: String,
    @ColumnInfo(name = "body") var body: String,
    @ColumnInfo(name = "date") var date: Date,
    @ColumnInfo(name = "updatedAt") var updatedAt: Date
): Serializable {
    @Ignore var isChecked: Boolean = false
}
